dot/state/grandpa.go

```diff
@@ -35,15 +35,15 @@ var (
 
 // GrandpaState tracks information related to grandpa
 type GrandpaState struct {
-	db         chaindb.Database
+	db         GetPutDeleter
 	blockState *BlockState
 
 	forcedChanges        *orderedPendingChanges
 	scheduledChangeRoots *changeTree
 }
 
 // NewGrandpaStateFromGenesis returns a new GrandpaState given the grandpa genesis authorities
-func NewGrandpaStateFromGenesis(db chaindb.Database, bs *BlockState,
+func NewGrandpaStateFromGenesis(db *chaindb.BadgerDB, bs *BlockState,
```

Comment:
This is left as an implementation on purpose, since it should be changed by issue #2981